src/Tools/dotnet-counters/CounterMonitor.cs

```diff
@@ -82,17 +83,51 @@ private void StopMonitor()
             _renderer.Stop();
         }
 
-        public async Task<int> Monitor(CancellationToken ct, List<string> counter_list, IConsole console, int processId, int refreshInterval)
+        private bool ValidateProcessIdAndDiagnosticsServerAddress()
+        {
+            if (string.IsNullOrEmpty(_diagnosticsServerAddress))
```

Comment:
Can we merge the two if statements?
```suggestion
            if (string.IsNullOrEmpty(_diagnosticsServerAddress) && if(_processId == 0))
```